internal/agent/delta/store.go

```diff
@@ -87,6 +89,58 @@ type Store struct {
 	CacheDir string
 	// NextIDMap stores the information about the available plugins
 	NextIDMap pluginIDMap
+	// stores time of last success submission of inventory to backend
+	lastSuccessSubmission time.Time
+}
+
+func (d *Store) LastSuccessSubmission() (time.Time, error) {
+
+	if !d.lastSuccessSubmission.Equal(time.Time{}) {
+		return d.lastSuccessSubmission, nil
+	}
+
+	content, err := ioutil.ReadFile(d.generateLastSuccessSubmissionFilePath())
```

Comment:
what could happen if we want to readFile that it's being written?

Should we lock it?

Reply:
file is read once at startup time, and then only in-memory reference is read